### PR TITLE
Added container-building best practices in the description and in the examples

### PR DESCRIPTION
**Pull request by mambelli**
Added container-building best practices in the description and in the examples
as suggested by @michmx

### CODE REVIEW

**Review by michmx — APPROVED** [review]

File: _episodes/04-building-containers.md [hunk 1]
```diff
@@ -19,6 +20,9 @@ defined in the documentation as the _Swiss army knife_ of container creation.
 The usual workflow is to prepare and test a container in a build environment (like your laptop),
 either with an interactive session or from a definition file,
 and then to deploy the container into a production environment for execution (as your institutional cluster).
+Interactive sessions are great to experiment and test your new container.
+If you want to distribute the container or use it in production, then we recommend to build it from a definition file, as described in the next episode,
```
Comment (klieret) [inline _episodes/04-building-containers.md:24]: this line should probably end with a '.'

File: _episodes/05-definition-files.md [hunk 2]
```diff
@@ -152,6 +158,13 @@ Here we have covered the basics with a few examples focused to HEP software.
 Check the [Apptainer docs](https://apptainer.org/docs/user/main/build_a_container.html) to see all the available
 options and more details related to the container creation.
 
+A few [best practices for your containers](https://apptainer.org/docs/user/1.0/definition_files.html#best-practices-for-build-recipes) to make them more usable, portable, and secure:
+1. Always install packages, programs, data, and files into operating system locations (e.g. not `/home`, `/tmp` , or any other directories that might get commonly binded on).
+1. Document your container. If your runscript doesn’t supply help, write a `%help` or `%apphelp` section. A good container tells the user how to interact with it.
+1. If you require any special environment variables to be defined, add them to the %environment and %appenv sections of the build recipe.
```
Comment (klieret) [inline _episodes/05-definition-files.md:164]: (could put `%env...`, `%appenv` in backticks for consistency)

**Comment by michmx** [comment]
Thanks a lot, @mambelli ! Fantastic to see it included in the chapter.

**Review by klieret — APPROVED** [review]

**Comment by michmx** [comment]
Ohhh very sorry! I was too fast... Didn't see @klieret comments.

**Comment by klieret** [comment]
I'll quickly push that myself :)

**Comment by klieret** [comment]
Done :)

**Comment by mambelli** [comment]
thanks for the fixes :)

**Pull request merged**